[OpenMS 2.1.0] PyOpenMS compilation error

Hi all.

`pyOpenMS` does not compile with this error:

```
pyopenms/pyopenms.cpp:54469:111: error: no matching function for call to 'OpenMS::InternalCalibration::applyTransformation(boost::shared_ptr<OpenMS::MSSpectrum<> >::element_type&, boost::shared_ptr<OpenMS::MZTrafoModel>::element_type&)'
     OpenMS::InternalCalibration::applyTransformation((*__pyx_v_spec->inst.get()), (*__pyx_v_trafo->inst.get()));
                                                               
```               

OpenMS 2.1.0
Fedora 26 64-bit
GCC 6.2.1
Boost 1.60.0
Full log: https://kojipkgs.fedoraproject.org//work/tasks/5839/16355839/build.log

--- a/src/openms/include/OpenMS/FILTERING/CALIBRATION/InternalCalibration.h
+++ b/src/openms/include/OpenMS/FILTERING/CALIBRATION/InternalCalibration.h
@@ -201,7 +201,7 @@ namespace OpenMS
                    const String& file_models_plot = "",
                    const String& file_residuals = "",
                    const String& file_residuals_plot = "",
-                   const QString& rscript_executable = "Rscript");
+                   const String& rscript_executable = "Rscript");
 
     /*
       @brief Transform a precursor's m/z

--- a/src/openms/source/FILTERING/CALIBRATION/InternalCalibration.cpp
+++ b/src/openms/source/FILTERING/CALIBRATION/InternalCalibration.cpp
@@ -265,8 +265,10 @@ namespace OpenMS
                                       const String& file_models_plot,
                                       const String& file_residuals,
                                       const String& file_residuals_plot,
-                                      const QString& rscript_executable)
+                                      const String& rscript_executable_)
   {
+    QString rscript_executable = rscript_executable_.toQString();
+
     // ensure sorting; required for finding RT ranges and lock masses
     if (!exp.isSorted(true))
     {

--- a/src/pyOpenMS/pxds/InternalCalibration.pxd
+++ b/src/pyOpenMS/pxds/InternalCalibration.pxd
@@ -7,6 +7,7 @@ from Peak1D cimport *
 from PeptideIdentification cimport *
 from ProgressLogger cimport *
 from Types cimport *
+from Precursor cimport *
 
 cdef extern from "<OpenMS/FILTERING/CALIBRATION/InternalCalibration.h>" namespace "OpenMS":
 
@@ -15,11 +16,11 @@ cdef extern from "<OpenMS/FILTERING/CALIBRATION/InternalCalibration.h>" namespac
         #    ProgressLogger
 
         InternalCalibration()      nogil except +
-        InternalCalibration(InternalCalibration) nogil except + 
-        
+        InternalCalibration(InternalCalibration) nogil except +
+
 
         Size fillCalibrants(MSExperiment[Peak1D,ChromatogramPeak],
-                            libcpp_vector[InternalCalibration_LockMass], 
+                            libcpp_vector[InternalCalibration_LockMass],
                             double tol_ppm,
                             bool lock_require_mono,
                             bool lock_require_iso,
@@ -29,23 +30,35 @@ cdef extern from "<OpenMS/FILTERING/CALIBRATION/InternalCalibration.h>" namespac
         Size fillCalibrants(libcpp_vector[PeptideIdentification], double) nogil except +
         CalibrationData getCalibrationPoints() nogil except +
         bool calibrate(MSExperiment[Peak1D,ChromatogramPeak],
-                       libcpp_vector[int], MZTrafoModel_MODELTYPE, double,
-                       bool, double, double, String, String) nogil except +
-                                     
+                       libcpp_vector[int],
+                       MZTrafoModel_MODELTYPE,
+                       double rt_chunk,
+                       bool use_RANSAC,
+                       double post_ppm_median,
+                       double post_ppm_MAD,
+                       String file_models,
+                       String file_models_plot,
+                       String file_residuals,
+                       String file_residuals_plot) nogil except +
+
+
+
 ## wrap static methods
 cdef extern from "<OpenMS/FILTERING/CALIBRATION/InternalCalibration.h>" namespace "OpenMS::InternalCalibration":
-    
-    void applyTransformation(MSSpectrum[Peak1D] & spec, 
-                                    MZTrafoModel & trafo) nogil except + # wrap-attach:InternalCalibration
-    void applyTransformation(MSExperiment[Peak1D, ChromatogramPeak] & exp, 
-                                    IntList& target_mslvl, MZTrafoModel& trafo) nogil except + # wrap-attach:InternalCalibration
+
+    void applyTransformation(libcpp_vector[Precursor]& pcs,
+                             MZTrafoModel& trafo) nogil except + # wrap-attach:InternalCalibration
+    void applyTransformation(MSSpectrum[Peak1D] & spec, IntList& target_mslvl,
+                             MZTrafoModel & trafo) nogil except + # wrap-attach:InternalCalibration
+    void applyTransformation(MSExperiment[Peak1D, ChromatogramPeak] & exp,
+                             IntList& target_mslvl, MZTrafoModel& trafo) nogil except + # wrap-attach:InternalCalibration
 
 cdef extern from "<OpenMS/FILTERING/CALIBRATION/InternalCalibration.h>" namespace "OpenMS::InternalCalibration":
-    
+
     cdef cppclass InternalCalibration_LockMass "OpenMS::InternalCalibration::LockMass":
         #InternalCalibration_LockMass() nogil except +
         InternalCalibration_LockMass(InternalCalibration_LockMass) nogil except + # wrap-ignore
-        InternalCalibration_LockMass(double mz_, int lvl_, int charge_) nogil except + 
+        InternalCalibration_LockMass(double mz_, int lvl_, int charge_) nogil except +
         double mz # m/z of the lock mass (incl. adducts)
         unsigned int ms_level # MS level where it occurs
         int charge # charge of the ion (to find isotopes)

--- a/src/pyOpenMS/pxds/InternalCalibration.pxd
+++ b/src/pyOpenMS/pxds/InternalCalibration.pxd
@@ -39,7 +39,8 @@ cdef extern from "<OpenMS/FILTERING/CALIBRATION/InternalCalibration.h>" namespac
                        String file_models,
                        String file_models_plot,
                        String file_residuals,
-                       String file_residuals_plot) nogil except +
+                       String file_residuals_plot,
+                       String rscript_executable) nogil except +
 
 
 

--- a/src/topp/InternalCalibration.cpp
+++ b/src/topp/InternalCalibration.cpp
@@ -356,7 +356,7 @@ protected:
                       getStringOption_("quality_control:models_plot"),
                       getStringOption_("quality_control:residuals"),
                       getStringOption_("quality_control:residuals_plot"),
-                      rscript_executable.toQString()))
+                      rscript_executable))
     {
       LOG_ERROR << "\nCalibration failed. See error message above!" << std::endl;
       return UNEXPECTED_RESULT;

